sevntu-checks/src/main/java/com/github/sevntu/checkstyle/filters/SuppressionAnnotationFilter.java

```diff
@@ -0,0 +1,428 @@
+        else {
+            startColumn = 0;
+        }
+        final int startLine = startNode.getLineNo();
+
+        final DetailAST endNode = getEndNode(node);
+        final int endLine = endNode.getLineNo();
+        final int endColumn = endNode.getColumnNo() + 1;
+        return new SuppressRange(startLine, startColumn, endLine, endColumn);
+    }
+
+    /**
+     * Get the rightmost lowest node, which is the end border element of the suppressed node.
+     * @param node
+     *         , the node to be suppressed.
+     * @return end border node.
+     */
+    private static DetailAST getEndNode(DetailAST node) {
+        DetailAST nextNode = node.getFirstChild();
+        final DetailAST result;
+        if (nextNode == null) {
+            result = node;
+        }
+        else {
+            DetailAST probe = nextNode;
+            while (true) {
+                probe = probe.getNextSibling();
+                if (probe == null) {
+                    break;
+                }
+                else {
+                    nextNode = probe;
+                }
+            }
+            result = getEndNode(nextNode);
+        }
+        return result;
+    }
+
+    /**
+     * A lot of checks return only number of the line, and number of the column is zeroed. It is
+     * supposed that code is already formatted and a chance that events which don't address to the
+     * annotated element are generated on one string with the annotated element can be neglected.
+     * The obvious exception is a parameter's definition.
+     * @param node
+     *         , the node to be suppressed.
+     * @return true
+     *          , if the start column index should not be set 0.
+     */
+    private static boolean columnIndexIsNecessary(DetailAST node) {
+        return node.getType() == TokenTypes.PARAMETER_DEF;
+    }
+
+    /**
+     * Checks if a check that generated the event, is not suppressed.
+     * @param event
+     *        , the event under audit.
+     * @return result of check.
+     */
+    private boolean matchesCheck(AuditEvent event) {
+        return checkRegexp.stream()
+            .map(pattern -> pattern.matcher(event.getSourceName()))
+            .anyMatch(Matcher::matches);
+    }
+
+    /**
+     * A SuppressRange is a POJO that holds start and end ranges of an element
+     * annotated with a suppressive annotation.
+     */
+    public static class SuppressRange {
+        /** Index of the first line of a suppressed range. */
+        private final int startLine;
+
+        /** Index of the first column of a suppressed range. */
+        private final int startColumn;
+
+        /** Index of the last line of a suppressed range. */
+        private final int endLine;
+
+        /** Index of the last column of a suppressed range. */
+        private final int endColumn;
+
+        /**
+         * Default constructor.
+         * @param startLine
+         *        , index of the first line of a suppressed range.
```

Comment:
Remove commas from start of parameter. These should be close to sentences.
Do this for all.